core/src/env/backend.rs

```diff
@@ -77,6 +77,27 @@ pub trait Env {
 
     /// Prints the given contents to the console log.
     fn println(&mut self, content: &str);
+
+    /// Returns the SHA2 256-bit hash of the given input.
+    fn hash_sha2_256(input: &[u8]) -> [u8; 32];
```

Comment:
Is there a specific reason that you do not use a mutable reference to array as an out parameter? Does the purer interface justify the additional copy of 32 bytes?

Reply:
You are right that we should provide a lower-level API for the functions implemented here. We can then implement some pure functions of top of those. Thanks.